### PR TITLE
Update docs and comments for 'federate enable/disable'

### PR DESCRIPTION
**Pull request by irfanurrehman**
As a follow up for #534 

cc @marun @gyliu513 @xunpan

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -179,24 +179,26 @@ It is possible to enable federation of any Kubernetes API type (including CRDs)
...
 
 The command will create CRDs for the primitives (template, placement, override) needed to
 federate the type, with names of the form `Federated<Kind>`, `Federated<Kind>Placement`, and
 `Federated<Kind>Override`.  The command will also create a `FederatedTypeConfig` in the
 federation system namespace with the group-qualified plural name of the target type.  A
-`FederatedTypeConfig` associates the primitive CRD types with the target type, and enables the
-primitive types to be propagated as the target type to member clusters.
+`FederatedTypeConfig` associates the primitive CRD types with the target kubernetes type, 
+enabling propagation of the federated resources of the given type to the member clusters.
+The name format used to name the `FederatedTypeConfig` is `<target kubernetes API type name>.<group name>`,
```
Comment (marun) [inline docs/userguide.md]: s/name format/format/

File: docs/userguide.md [hunk 2]
```diff
@@ -208,21 +210,28 @@ It is possible to disable propagation of a type that is configured for propagati
...
 ```bash
-kubefed2 federate disable <FederatedTypeConfig name>
+kubefed2 disable <FederatedTypeConfig Name>
 ```
 
-This command will set the `propagationEnabled` field in the `FederatedTypeConfig` to `false`,
-which will prompt the sync controller for the type to be stopped.
+This command will set the `propagationEnabled` field in the `FederatedTypeConfig` 
+associated with this target API type to `false`, which will prompt the sync 
+controller to stop sync for this target API type.
```
Comment (marun) [inline docs/userguide.md]: s/to stop sync for this target API type/for the target API type to be stopped/

I think it's important to note that the controller will be stopped, not just that propagation will be stopped.  The difference between the 2 statements has operational consequences.

File: docs/userguide.md [hunk 3]
```diff
@@ -208,21 +210,28 @@ It is possible to disable propagation of a type that is configured for propagati
...
 
 ```bash
-kubefed2 federate disable <FederatedTypeConfig name> --delete-from-api
+kubefed2 disable <FederatedTypeConfig Name> --delete-from-api
 ```
 
 **WARNING: All primitive custom resources for the type will be removed by this command.**
 
+***Note***: The NAME needed in `disable` command is the name of the FederatedTypeConfig
+		created by `kubefed2 enable` for the `target kubernetes API type`. The name format used
```
Comment (marun) [inline docs/userguide.md]: I don't think it should be necessary to repeat the documentation of the name form here.

File: pkg/kubefed2/federate/disable.go [hunk 4]
```diff
@@ -47,11 +47,19 @@ var (
...
+		# Disable propagation of the kubernets API type 'Deployment', named
+		in FederatedTypeConfig as 'deployment.apps'
+		kubefed2 disable deployment.apps
+
+		# Disable propagation of the kubernets API type 'Deployment', named
+		in FederatedTypeConfig as 'deployment.apps', and delete corresponding
+		Federated API resources
+		kubefed2 disable deployment.apps --delete-from-api
+
+		Note: The NAME needed in disable command is the name of the FederatedTypeConfig
```
Comment (marun) [inline pkg/kubefed2/federate/disable.go]: I don't think it should be necessary to document this here.  The name of an ftc can be discovered from the API - it doesn't need to be invented.

**Comment by irfanurrehman** [comment]
@marun Thanks for taking a look, update!

File: pkg/kubefed2/federate/disable.go [hunk 5]
```diff
@@ -47,11 +47,14 @@ var (
 		--host-cluster-context flag otherwise.`
 
 	disable_example = `
-		# Disable propagation of the Service type
-		kubefed2 federate disable Service
-
-		# Disable propagation of the Service type and delete API resources
-		kubefed2 federate disable Service --delete-from-api`
+		# Disable propagation of the kubernets API type 'Deployment', named
```
Comment (marun) [inline pkg/kubefed2/federate/disable.go]: My apologies for not noticing this on my first pass: s/kubernets/kubernetes/ (also appears on line 54)

**Comment by marun** [comment]
Thank you for responding to my comments.  There's a minor spelling mistake to fix and then I think this is ready.

**Comment by irfanurrehman** [comment]
> Thank you for responding to my comments. There's a minor spelling mistake to fix and then I think this is ready.

@marun updated! Apologies for not catching that myself the first time.

**Comment by marun** [comment]
Thanks!

/lgtm

**Pull request merged**